### ISSUE TITLE
Inndata-type tydeliggjort - mobil

### ISSUE THREAD

**Issue by Janingetjosvoll** [Altinn/app-frontend-react#2374]
### Description

UU-krav 611. WCAG 2.1: 1.3.5. Det må være programatisk mulig å avgjøre hva det forventes at brukeren skriver inn i et felt, eller hva meningen med informasjonen er. Dette innebærer å bruke logiske ledetekster, definere indata-type og å inkludere autocomplete og lagret skjemainnhold der det passer seg.
For eksempel er et felt hvor en kun skal kunne skrive inn tall, kodet slik at tall-tastaturet vises på mobilskjerm. Datofeltet har f.eks. ikke "infotype", men du kan kun taste inn tall.
 Dette fungerer ikke på mobil per nå.

**Comment by olemartinorg** [comment]
We should look into the autocomplete property, fix that (see #2370) and investigate how this works on mobile when properly set up. If we need to require an autocomplete property that will be a breaking change that we should await introducing until app-frontend v5.

**Comment by kentaurse** [comment]
@Janingetjosvoll 
Yes, I’m planning to address this by first reviewing the current input components to ensure they properly define input types and use appropriate autocomplete attributes. 
I’ll start by fixing the autocomplete property (as noted in #2370), and then test how the inputs behave on mobile devices—particularly ensuring that the correct keyboard (e.g., numeric) appears where expected.
If it turns out that we need to enforce autocomplete across all fields, I’ll document it clearly and suggest handling it as a breaking change in app-frontend v5, in line with the team's guidance.

**Comment by olemartinorg** [comment]
Jeg flyttet den inn i TODO-kolonna nå, så da er den på kortlista iallefall. Såvidt jeg kan se er #2370 løst, så nå skal det (igjen) være mulig å sette autocomplete-egenskapen på input-komponenter. Vi kan som sagt ikke gjøre den egenskapen påkrevd helt umiddelbart, så om det trengs må vi nok få på plass en ny oppgave som kan gjøres til v5.

Regner da med at problemet her i praksis omhandler å sette riktig autocomplete-egenskap på datokomponenten. Jeg ser du nevnte det som et eksempel - har du oppdaget flere eksempler?

**Comment by tvs-brreg** [comment]
For at autocomplete skal fungere godt må da også felttypene være rette slik at den vet hva den skal foreslå.

Dette er feks en datepicker, den kunne hatt `type= "date"`
![Image](https://github.com/user-attachments/assets/efdb1202-7479-48d8-b008-b2ae2eab51c2)

Dette er et navnfelt og kunne hatt `type= "name"` eller om vi vil være enda mer spesifikk `type= "family-name"` siden vi skal ha inn etternavnet sammen med fødselsnummer
![Image](https://github.com/user-attachments/assets/c0157d5e-c7ed-4f68-9737-c5effe8c5554)

**Comment by olemartinorg** [comment]
Skjønner. For den førstnevnte, som er av typen `Datepicker` i layout-filen, tenker jeg også den burde hatt `type='date'` eller lignende.

For sistnevnte (`rettighetshaver-utenlandskPerson-fulltNavn`) finnes det ikke en gyldig attributt med `type= "name"`, men du kan sette `"autocomplete": "family-name"` i layout-filen allerede (dette ble som nevnt fikset i #2370).

**Comment by olemartinorg** [comment]
En ekstra ting jeg kom på nå er at vi nok kan gjøre noe smarter i oppslagskomponentene våre også:

- `PersonLookup` burde nok ha autocomplete-attributt for å indikere at man er på jakt etter et etternavn
- `OrganisationLookup` burde potensielt også ha et passende autocomplete-attributt
- `Address` kan nok også få bedre defaults

**Comment by olemartinorg** [comment]
@Janingetjosvoll Hei! Endringene vi kom med i forbindelse med denne saken er listet opp i #3391, og disse er lansert i https://github.com/Altinn/app-frontend-react/releases/tag/v4.20.0-preview.1 - og kommer dermed med i neste minor-versjon (v4.20.0).

Hvis du har mulighet for å ta en titt, setter jeg stor pris på tilbakemeldinger. Både om ting fungerer bedre etter dette, eller om du finner andre problemer jeg ikke oppdaget. 🙏
